Sources/TuistGenerator/Mappers/ModuleMapMapper.swift

```diff
@@ -0,0 +1,123 @@
+import Foundation
+import TuistCore
+import TuistGraph
+import TSCBasic
+
+/// Mapper that maps the `MODULE_MAP` build setting to the `-fmodule-map-file` compiler flags.
+/// It is required to avoid embedding the module map into the frameworks during cache operations, which would make the framework not portable, as
+/// the modulemap could contain absolute paths.
+public final class ModuleMapMapper: WorkspaceMapping {
+    private static let modulemapFileSetting = "MODULEMAP_FILE"
+    private static let otherSwiftFlagsSetting = "OTHER_SWIFT_FLAGS"
+
+    public init() {}
+
+    public func map(workspace: WorkspaceWithProjects) throws -> (WorkspaceWithProjects, [SideEffectDescriptor]) {
+        var targetToModuleMaps: [TargetID: Set<AbsolutePath>] = [:]
+        workspace.projects.forEach { project in
+            project.targets.forEach { target in
+                Self.dependenciesModuleMaps(workspace: workspace, project: project, target: target, targetToModuleMaps: &targetToModuleMaps)
+            }
+        }
+
+        var mappedWorkspace = workspace
+        for p in 0 ..< workspace.projects.count {
+            var mappedProject = workspace.projects[p]
+            for t in 0 ..< mappedProject.targets.count {
+                var mappedTarget = mappedProject.targets[t]
+                let targetID = TargetID(projectPath: mappedProject.path, targetName: mappedTarget.name)
+                var mappedSettingsDictionary = mappedTarget.settings?.base ?? [:]
+                guard mappedSettingsDictionary[Self.modulemapFileSetting] != nil || targetToModuleMaps[targetID] != nil else { continue }
+
+                if mappedSettingsDictionary[Self.modulemapFileSetting] != nil {
+                    mappedSettingsDictionary[Self.modulemapFileSetting] = nil
+                }
+
+                if let dependenciesModuleMaps = targetToModuleMaps[targetID] {
```

Comment:
can e.g. the contents of this `if let` be moved a to a `private function`? I feel like this `map` function is kind of hard to digest. At the very least, I'd add some small comments where you would see appropriate